bitflip/bitflip.rb

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env ruby
+
+require 'net/http'
+require 'resolv'
+
+def pad_with_zeros(string)
+	padded = string
+	until padded.length == 8
+		padded = "0" + padded
+	end
+	return padded
+end
+
+def is_valid_domain_char(original, char)
+	return false if char == original - 32
+	return (65 <= char && char <= 90) || (97 <= char && char <= 122) || (char == 45)
```

Comment:
You forgot to include the ASCII range for numbers `[0-9]`.
These can be in domains.

Also do you need both upper and lowercase? I assume that your browser, for example, would also just `downcase!` everything before submitting it to DNS. Perhaps even a machines kernal will do that before a network call? But probably good to account for the case where you bitflip from `a` into `B` while the packet is traveling to DNS - I assume that DNS will then downcase everything.